stats/stats.go

```diff
@@ -383,41 +392,43 @@ func (acc *numericAcc) Map() map[string]interface{} {
 		return map[string]interface{}{"count": 0}
 	}
 	m := map[string]interface{}{
+		"mean":  acc.mean,
 		"count": acc.count,
 		"min":   acc.min,
 		"max":   acc.max,
 	}
 
-	if acc.unique != 0 {
-		m["unique"] = acc.unique
+	if acc.median != maxFloat {
+		m["median"] = acc.median
 	}
 
-	if acc.frequencies != nil {
-		// need to convert keys to strings b/c many serialization formats aren't
-		// down with numeric map keys
-		strFrq := map[string]int{}
-		for fl, freq := range acc.frequencies {
-			strFrq[strconv.FormatFloat(fl, 'f', -1, 64)] = freq
+	if acc.histogram != nil {
+		m["histogram"] = map[string][]float64{
+			"bins":        acc.dividers,
+			"frequencies": acc.histogram,
 		}
-		m["frequencies"] = strFrq
 	}
 
 	return m
 }
 
 // Close finalizes the accumulator
 func (acc *numericAcc) Close() {
-	if acc.frequencies != nil {
-		// determine unique values
-		for key, freq := range acc.frequencies {
-			if freq == 1 {
-				acc.unique++
-				delete(acc.frequencies, key)
-			}
-		}
-		if len(acc.frequencies) == 0 {
-			acc.frequencies = nil
-		}
+	// finalize avg
+	acc.mean = acc.mean / float64(acc.count)
+
+	if len(acc.histogram) > 0 {
+		acc.median = acc.histogram[len(acc.histogram)/2]
```

Comment:
This is grabbing the middle entry before the histogram is sorted. Also, there is no catch in place for if the histogram list is even, in which we should be averaging the two middle entries,